Push code from the nf-core dev branch to the nf-core master for the first release 1.0.0

saima-tithi-stjude opened the pull request:
- Push code from the nf-core dev branch to the nf-core master for the first pipeline release 1.0.0.
- Bumped version from 1.0.0dev to 1.0.0

saima-tithi-stjude committed:
Initial commit with test working

saima-tithi-stjude committed:
updated config

saima-tithi-stjude committed:
Updated code to run the test-datasets in nf-core

saima-tithi-stjude committed:
Updated schema

saima-tithi-stjude committed:
Updated README

saima-tithi-stjude committed:
Update README

saima-tithi-stjude committed:
Updated docs files

saima-tithi-stjude committed:
Update docs files

saima-tithi-stjude committed:
deleted unused module codes

saima-tithi-stjude committed:
Updated schema and config

saima-tithi-stjude committed:
updated test dataset path in config

saima-tithi-stjude committed:
added workflow diagram

saima-tithi-stjude committed:
updated readme

saima-tithi-stjude committed:
updated base conf

saima-tithi-stjude committed:
updated schema and updated CADD file path

saima-tithi-stjude committed:
updated schema

saima-tithi-stjude committed:
updated config

saima-tithi-stjude committed:
Update schema

saima-tithi-stjude committed:
Merge pull request #1 from nf-core/dev

Updated schema file

saima-tithi-stjude committed:
Merge pull request #1 from saima-tithi-stjude/dev

Merge pull request #1 from nf-core/dev

saima-tithi-stjude committed:
fixed indentation

saima-tithi-stjude committed:
branch 'dev' of https://github.com/saima-tithi-stjude/CoCoRV_nf-core into dev
Merge commits

saima-tithi-stjude committed:
Merge pull request #2 from nf-core/dev

Added option for providing normalized files and fixed indentation

saima-tithi-stjude committed:
Merge pull request #3 from saima-tithi-stjude/dev

Merge pull request #2 from nf-core/dev

saima-tithi-stjude committed:
Added sex-stratified analysis and DNAnexus options

saima-tithi-stjude committed:
Fixed linting errors

saima-tithi-stjude committed:
Fixed schema file

saima-tithi-stjude committed:
Updated docs

saima-tithi-stjude committed:
updated default vep plugin list

saima-tithi-stjude committed:
updated process label for annotate_vep

saima-tithi-stjude committed:
added aws healthomics cmds and readme

saima-tithi-stjude committed:
added aws testrun param example

saima-tithi-stjude committed:
updated aws testrun

saima-tithi-stjude committed:
fixed linting

saima-tithi-stjude committed:
Updated aws healthomics readme

saima-tithi-stjude committed:
fixed aws healthomics readme

saima-tithi-stjude committed:
updated readme and aws docs

saima-tithi-stjude committed:
updated aws readme

saima-tithi-stjude committed:
updated aws readme and pipeline image

saima-tithi-stjude committed:
updated readme

saima-tithi-stjude committed:
Merge pull request #3 from nf-core/dev

merge all commits

saima-tithi-stjude committed:
Merge pull request #4 from saima-tithi-stjude/dev

Merge pull request #3 from nf-core/dev

saima-tithi-stjude committed:
updated readme and workflow image

saima-tithi-stjude committed:
Merge branch 'dev' of https://github.com/saima-tithi-stjude/CoCoRV_nf-core into dev

saima-tithi-stjude committed:
updated aws readme

saima-tithi-stjude committed:
bumped version to 1.0.0 and updated changelog

saima-tithi-stjude committed:
bumped version back to dev

saima-tithi-stjude committed:
Added dnanexus readme

saima-tithi-stjude committed:
Updated acanconfig param

saima-tithi-stjude committed:
updated dnanexus readme

saima-tithi-stjude committed:
updated dnanexus docs

saima-tithi-stjude committed:
updated dnanexus readme and conda paths

saima-tithi-stjude committed:
fixed container options

saima-tithi-stjude committed:
updated postCheck module

saima-tithi-stjude committed:
updated for dnanexus disk storage

saima-tithi-stjude committed:
fixed bug in annotate_vep

saima-tithi-stjude committed:
updated main.nf to ensure pipeline_info is copied in outdir on aws

saima-tithi-stjude committed:
updated aws testrun params file

saima-tithi-stjude committed:
fixed config file

saima-tithi-stjude committed:
updated nextflow.config for saving report files for aws omics

saima-tithi-stjude committed:
updated aws testrun params

saima-tithi-stjude committed:
updated docker image for r version

nvnieuwk commented on subworkflows/local/utils_nfcore_rarevariantburden_pipeline/main.nf:
```diff
@@ -68,33 +67,9 @@ workflow PIPELINE_INITIALISATION {
     //
     validateInputParameters()
 
-    //
-    // Create channel from input file provided through params.input
-    //
-
-    Channel
-        .fromList(samplesheetToList(params.input, "${projectDir}/assets/schema_input.json"))
-        .map {
-            meta, fastq_1, fastq_2 ->
-                if (!fastq_2) {
-                    return [ meta.id, meta + [ single_end:true ], [ fastq_1 ] ]
-                } else {
-                    return [ meta.id, meta + [ single_end:false ], [ fastq_1, fastq_2 ] ]
-                }
-        }
-        .groupTuple()
-        .map { samplesheet ->
-            validateInputSamplesheet(samplesheet)
-        }
-        .map {
-            meta, fastqs ->
-                return [ meta, fastqs.flatten() ]
-        }
-        .set { ch_samplesheet }
-
```
Why did you remove this code? This is used to validate and dynamically read in the samplesheet file. I saw that you use `.splitCsv` instead which is less stable than the `samplesheetToList` function of the `nf-schema` plugin. 

See the documentation to learn more about this: https://nextflow-io.github.io/nf-schema/latest/samplesheets/samplesheetToList/

This is a must-have in nf-core pipelines and the only exceptions are made in very specific cases (this doesn't seem like such a case IMO)

Feel free to reach out to me if you need any help :)

nvnieuwk commented on subworkflows/local/utils_nfcore_rarevariantburden_pipeline/main.nf:
```diff
@@ -112,11 +87,11 @@ workflow PIPELINE_COMPLETION {
     monochrome_logs // boolean: Disable ANSI colour codes in log output
     hook_url        //  string: hook URL for notifications
-    multiqc_report  //  string: Path to MultiQC report
+    association_res  //  string: Path to MultiQC report
+    qqplot
```
Can you fix the comment next to `association_res` and add one next to `qqplot`?

nvnieuwk reviewed (changes requested):
After taking a quick look I noticed a lot of clashes with the nf-core guidelines for this pipeline. I already commented on some but I think it might be better if you read through the guidelines yourself and see which parts could be fixed. Feel free to message me or ask me any questions in this PR.

You can find all the guidelines here: https://nf-co.re/docs/guidelines/pipelines/overview

Please also take a look at module guidelines: https://nf-co.re/docs/guidelines/components/modules

saima-tithi-stjude committed:
fixed for running on dnanexus

saima-tithi-stjude committed:
fixed type for annovarFolder param

nvnieuwk commented:
@saima-tithi-stjude any updates on this?

saima-tithi-stjude committed:
updated postcheck script

saima-tithi-stjude committed:
added sort for the gene variant list

saima-tithi-stjude committed:
added cocorv source and run conda profile

saima-tithi-stjude committed:
Fix nextflow.config: avoid cross-block self-reference in manifest/validation banner text

saima-tithi-stjude committed:
Fix nextflow.config: remove def statements, use literal text (no cross-block refs)

saima-tithi-stjude committed:
Bump nf_core_version to 3.3.2 to fix broken bundled template in files_unchanged check

saima-tithi-stjude committed:
Bump nf_core_version to 4.0.2 (3.3.2 still had the broken bundled template)

nf-core-bot committed:
Remove vulnerable PR-comment artifact pattern

These files conflicted with the patch and were reset wholesale to the nf-core/tools 4.0.3 template version, on the basis that they carry no pipeline-specific content. If this branch had local edits in them, those edits are gone — please review: .github/workflows/branch.yml .github/workflows/template-version-comment.yml

saima-tithi-stjude committed:
Replace splitCsv() with schema-validated samplesheetToList() for all file-list params

saima-tithi-stjude committed:
Apply prettier formatting to new schema files

saima-tithi-stjude committed:
Fix missing executable permission on bin/ scripts

saima-tithi-stjude committed:
Remove def statement in test.config for Nextflow strict-config compatibility

saima-tithi-stjude committed:
Remove workflow.onComplete HealthOmics debug block

mashehu committed:
Merge pull request #10 from nf-core/patch

Remove vulnerable PR-comment artifact pattern

saima-tithi-stjude committed:
updated environment-r.yaml

saima-tithi-stjude committed:
Fixed the linting errors

saima-tithi-stjude committed:
Merge branch 'dev' of https://github.com/saima-tithi-stjude/CoCoRV_nf-core into dev

saima-tithi-stjude committed:
Restore accidentally overwritten utils_nfcore_pipeline subworkflow

saima-tithi-stjude committed:
Fix missing executable permission on vendored assets/cocorv utility scripts

saima-tithi-stjude committed:
Add missing test_full profile entry to fix nf-core pipelines download

saima-tithi-stjude committed:
Restore lost files_exist/files_unchanged lint exclusions, add nextflow_config exclusion for params.input

saima-tithi-stjude committed:
fixed ro-crate-metadata.json for linting err

saima-tithi-stjude committed:
Add minimal nf-test scaffolding

saima-tithi-stjude committed:
Add missing get-shards composite action

saima-tithi-stjude committed:
Add bioconda channel to nf-test conda setup

saima-tithi-stjude committed:
Align nf-test conda setup with proven-working ci.yml configuration

saima-tithi-stjude committed:
Add modules_testdata_base_path; exclude versions.yml nf_test_content check
